packages/admin/admin/src/rowActions/RowActionsListItem.tsx

```diff
@@ -4,30 +4,38 @@ import * as React from "react";
 
 import { CommonRowActionItemProps } from "./RowActionsItem";
 
-export type RowActionsListItemComponentsProps = React.PropsWithChildren<{
+export type RowActionsListItemComponentsProps<MenuItemComponent extends React.ElementType = "li"> = React.PropsWithChildren<{
     listItemIcon?: Partial<ListItemIconProps>;
     listItemText?: Partial<ListItemTextProps>;
-    menuItem?: Partial<MenuItemProps>;
+    menuItem?: Partial<MenuItemProps<MenuItemComponent> & { component: MenuItemComponent }>;
 }>;
 
-export interface RowActionsListItemProps extends CommonRowActionItemProps {
+export interface RowActionsListItemProps<MenuItemComponent extends React.ElementType = "li"> extends CommonRowActionItemProps {
     textSecondary?: React.ReactNode;
     endIcon?: React.ReactNode;
-    componentsProps?: RowActionsListItemComponentsProps;
+    componentsProps?: RowActionsListItemComponentsProps<MenuItemComponent>;
     children?: React.ReactNode;
 }
 
-export const RowActionsListItem = React.forwardRef<HTMLLIElement, RowActionsListItemProps>(function RowActionsListItem(props, ref) {
+const RowActionsListItemNoRef = <MenuItemComponent extends React.ElementType>(
+    props: RowActionsListItemProps<MenuItemComponent>,
+    ref: React.ForwardedRef<HTMLLIElement>,
+) => {
     const { icon, children, textSecondary, endIcon, componentsProps = {}, ...restMenuItemProps } = props;
     const { listItemIcon: listItemIconProps, listItemText: listItemTextProps, menuItem: menuItemProps } = componentsProps;
+
     return (
         <MenuItem ref={ref} {...restMenuItemProps} {...menuItemProps}>
             {icon !== undefined && <ListItemIcon {...listItemIconProps}>{icon}</ListItemIcon>}
             {children !== undefined && <ListItemText primary={children} secondary={textSecondary} {...listItemTextProps} />}
             {Boolean(endIcon) && <EndIcon>{endIcon}</EndIcon>}
         </MenuItem>
     );
-});
+};
+
+export const RowActionsListItem = React.forwardRef(RowActionsListItemNoRef) as <MenuItemComponent extends React.ElementType>(
```

Comment:
This as cast and the additional "NoRef" component seem wrong. Could you please try to fix the typing without them?

Reply:
I tried really long to do it cleaner but didn't find a way. If you like I can take a look again 😅

Reply:
Seems like there really is no nice way to do this (https://fettblog.eu/typescript-react-generic-forward-refs/#option-1%3A-type-assertion, https://stackoverflow.com/a/58473012)

Reply:
So @johnnyomair are you alright with me leaving it like this?

Reply:
I'd like to try this myself once you've resolved the conflicts

Reply:
I've tried this locally and notice that your `component` implementation isn't correct anyway 😅 Overriding the component should add the props required by the component to the props, but we're only adding component here. I'd suggest we remove the generic altogether and add a `@ts-expect-error` instead, as it seems that MUI doesn't plan to fix the issue: https://github.com/mui/material-ui/issues/33799

Reply:
I don't quite get what you mean exactly with 'we're only adding component here'😅

Reply:
This: `Partial<MenuItemProps<MenuItemComponent> & { component: MenuItemComponent }>`. MUI achieves component overridability by using the `OverrideProps` helper, which also adds the component's props (for instance, `href` and `target` for `a`) to the resulting type

Reply:
I just tried it and it works as you can see in the screen rec. I didn't change anything in the code.

https://github.com/vivid-planet/comet/assets/56400587/6d0c835e-3f46-4fb7-ba86-484eb20a1875